aea/crypto/cosmos.py

```diff
@@ -126,30 +128,34 @@ def sign_transaction(self, transaction: Any) -> Any:
         :param transaction: the transaction to be signed
         :return: signed transaction
         """
-        message_str = json.dumps(transaction, separators=(",", ":"), sort_keys=True)
-        message_bytes = message_str.encode("utf-8")
-        signed_transaction = self.sign_message(message_bytes)
+        transaction_str = json.dumps(transaction, separators=(",", ":"), sort_keys=True)
+        transaction_bytes = transaction_str.encode("utf-8")
+        signed_transaction = self.sign_message(transaction_bytes)
         return signed_transaction
 
     def recover_message(
         self, message: bytes, signature: str, is_deprecated_mode: bool = False
-    ) -> Address:
+    ) -> Tuple[Address, ...]:
         """
-        Recover the address from the hash.
+        Recover the addresses from the hash.
 
         :param message: the message we expect
         :param signature: the transaction signature
         :param is_deprecated_mode: if the deprecated signing was used
-        :return: the recovered address
+        :return: the recovered addresses
         """
         signature_b64 = base64.b64decode(signature)
         verifying_keys = VerifyingKey.from_public_key_recovery(
             signature_b64, message, SECP256k1, hashfunc=hashlib.sha256,
         )
-        # TODO: pick correct key
-        public_key = verifying_keys[0].to_string("compressed").hex()
-        address = self.get_address_from_public_key(public_key)
-        return address
+        public_keys = [
+            verifying_key.to_string("compressed").hex()
+            for verifying_key in verifying_keys
+        ]
+        addresses = [
+            self.get_address_from_public_key(public_key) for public_key in public_keys
+        ]
+        return tuple(addresses)
```

Comment:
Unlike in ethereum, in cosmos any signature+message can be matched to two public keys/addresses since the `s` is not restricted.

Reply:
ok